pkg/token/execCredentialPlugin.go

```diff
@@ -37,7 +38,11 @@ func New(o *Options) (ExecCredentialPlugin, error) {
 		if err := json.Unmarshal([]byte(env), &execCredential); err != nil {
 			return nil, fmt.Errorf("cannot convert to ExecCredential: %w", err)
 		}
-		if !execCredential.Spec.Interactive && o.LoginMethod == DeviceCodeLogin {
+		klog.V(10).Infof("KUBERNETES_EXEC_INFO is: %s", env)
+
+		if strings.Contains(env, `"spec":{}`) {
+			fmt.Fprintln(os.Stderr, "Your current kubectl version is below 1.22.0 and it is highly recommended to use 1.22.0 and above.For upgrading, please refer to https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/")
```

Comment:
i think we can save the message to not annoy people :)

Reply:
Done in commit a351ac4. Please check it out. Thanks!